### DISCUSSION TITLE
Milvus Search Failure: "vector dimension mismatch/no available shard delegator found: service unavailable"

### DISCUSSION THREAD

**Discussion by Seanthelast** [milvus-io/milvus#30686] [Q&A and General discussion]
Hello,
I've had luck with inserting and searching in Milvus before though this time I've been getting a weird error. I have inserted only one collection into my Milvus-Lite database, which consists of vector embeddings done with Llama 2 7b, all of dimension of 200. Insertion happens with no errors. When trying to search using a query, I get a search error, that goes along the lines of "vector dimension mismatch" and "no available shard delegator found: service unavailable". I have searched several solutions for this though I cannot seem to find anything about it. People have gotten a similar error before, though it is not exactly the same and the causes are different. At this point, I am curious if the fact that a lot of my vector embeddings have added zero-paddings that I manually added to each in order to make them all of equal length 200. Besides that, I am not sure. 
I've uploaded an image if the first couple lines of the error as well as the search code, if you need anything else to help diagnose the problem, let me know and I'll send it.
![image](https://github.com/milvus-io/milvus/assets/89895021/f24d796d-6393-4072-b0ae-894af283c9df)
![image](https://github.com/milvus-io/milvus/assets/89895021/395ce690-5cc4-4e11-b7ce-2147b144ef2a)

**Comment by yhmo** [comment]
What is the output of `tokenizer.encode()` ?

The parameter "data" of collection.search() is a list of vectors, each vector is a list of float values, and the length of each list should be equal to the dimension.

**Reply by Seanthelast** [reply]
The output would be an embedding the size of the query, which would be less than 200. 
So the query has to be equal size to the vectors of the DB? I'm a little confused, as I've never had to do that before and I've still been able to get search results, especially since a one sentence query is going to be smaller than an embedding of several sentences. I could add zero padding to the query embedding but I just want to understand why this would be necessary.

**Reply by yhmo** [reply]
ANNS(Approximate Nearest Neighbor Search) is based on distance/similarity metric between embeddings. Euclidean distance and inner product is the two popular metrics.
Assume there are two embeddings:
vec_1 = [a1, a2, a3, ... a100]
vec_2 = [b1, b2, b3, ... b10]

Euclidean distance = `sqrt((a1-b1)^2 + (a2-b2)^2 + ...... + (a100-b100)^2)`
Inner product = `a1*b1 + a2*b2 + ...... + a100*b100`

So, we can see the algorithm requires the two embeddings have the same dimension.

For binary vector, HAMMIN is the popular metric, it compares two vectors bit by bit, also resuires the two vectors have the same bit length.
